controllers/comment-controllers.js

```diff
@@ -0,0 +1,43 @@
+const { deleteComment, addComment } = require("../models/comment-models");
+const {
+  checkCommentIdExists,
+  checkIdExists,
+  checkUserNameExist,
+} = require("../models/article-parameters-exists-models");
+
+const postComment = (request, response, next) => {
+  const { body } = request.body;
+  const { username } = request.body;
+  const { article_id } = request.params;
+  if (body === undefined || username === undefined) {
+    response.status(400).send({ msg: "missing required field" });
+  }
+
+  Promise.all([checkIdExists(article_id), checkUserNameExist(username)])
```

Comment:
We don't need to check either of these things are both are foreign keys. If either does not exist, PSQL will throw a foreign key violation error

Reply:
Same as above, I've separated the message responses based on whether its specifically the username or id that is invalid, not sure if I can do that with a single error handler